examples/data_types_and_io/Dockerfile

```diff
@@ -17,8 +18,10 @@ ENV VENV /opt/venv
 RUN python3 -m venv ${VENV}
 ENV PATH="${VENV}/bin:$PATH"
 
-RUN pip install flytekit pandas
-RUN pip install torch --index-url https://download.pytorch.org/whl/cpu
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/.cache/pip,id=pip \
+    pip install flytekit pandas pyarrow
+RUN --mount=type=cache,sharing=locked,mode=0777,target=/root/.cache/pip,id=pip \
```

Comment:
the use of the `sharing=locked` option is intentional, right?

Reply:
Yup, the default is `sharing=shared`. Last time I looked into pip's source, it can end up corrupting the cache if there are multiple writers. For docker, multiple writes can happen if there are multiple docker builds going on at the same time.

with `sharing=locked`, there can only be one builder using the cache.

If we want to keep this PR very simple, then the alternative is `pip install --no-cache-pip`.